### ISSUE TITLE
All urls return 404 page after upgrade to 10.1.0 when runtime mode set to Production

### ISSUE THREAD

**Issue by teeto** [umbraco/Umbraco-CMS#12792]
### Which *exact* Umbraco version are you using? For example: 9.0.1 - don't just write v9

10.1.0

### Bug summary

All urls return 404 page after upgrade to 10.1.0 when runtime mode is set to:
`"Runtime": {
       "Mode": "Production"
      }`

if i comment these lines from the appsettings it works ok.

### Steps to reproduce

Upgrade from 10.0.1 to 10.1.0
Set runtime mode to production.

**Comment by nul800sebastiaan** [comment]
I can reproduce. Updated `appSettings.json` to the following:

```json
{
  "$schema": "./appsettings-schema.json",
  "Serilog": {
    "MinimumLevel": {
      "Default": "Information",
      "Override": {
        "Microsoft": "Warning",
        "Microsoft.Hosting.Lifetime": "Information",
        "System": "Warning"
      }
    }
  },
  "Umbraco": {
    "CMS": {
      "Runtime": {
        "Mode": "Production"
      },
      "Global": {
        "Id": "a770bdfc-0fe8-45df-941b-dda79686d7da",
        "SanitizeTinyMce": true,
        "UseHttps": true
      },
      "ModelsBuilder": {
        "ModelsMode": "Nothing"
      },
      "Content": {
        "ContentVersionCleanupPolicy": {
          "EnableCleanup": true
        }
      },
      "WebRouting": {
        "UmbracoApplicationUrl": "https://localhost:5001/"
      }
    }
  },
  "ConnectionStrings": {
    "umbracoDbDSN": "Data Source=(localdb)\\MSSQLLocalDB;AttachDbFilename=|DataDirectory|\\Umbraco.mdf;Integrated Security=True",
    "umbracoDbDSN_ProviderName": "Microsoft.Data.SqlClient"
  }
}
```

Then did a publish: `dotnet publish -c Release -o .\publish`. In the `publish` directory ran the `MyProject.exe` and went to https://localhost:5001/ where I ran the installer and then the starter kit installer kicked in (used the https://www.nuget.org/packages/clean starter kit). All of the URLs present with "Page not found". 

![image](https://user-images.githubusercontent.com/304656/183364602-57d8223a-bad6-4d89-8e87-74ec4fee6388.png)

**Comment by ronaldbarendse** [comment]
Hi @teeto, you probably still have the `RazorCompileOnBuild` and `RazorCompileOnPublish` properties in your project file set to `false`... Because `Production` runtime mode disables Razor runtime compilation, views are only available when they're compiled on build/publish, hence the need to remove these properties (or explicitly set them to `true` - the default value). If you don't, Umbraco returns a 404 because it can't find the template.

The [documentation mentions this](https://our.umbraco.com/Documentation/Fundamentals/Setup/Server-Setup/runtime-modes):
- _Although this setting affects how Umbraco behaves at runtime, some modes have prerequisites on how the project is built/published. Make sure to read the descriptions of each mode before changing this setting from the default `BackofficeDevelopment` mode [...]_
- Development mode:
  - _It is a recommended prerequisite if you want to use the Production mode in your production environment._
  - _You can remove the following properties from your csproj project file to enable the compilation of Razor views (which also ensures your views do not contain compilation errors and is a prerequisite for enabling Production mode):_
    ```xml
    <RazorCompileOnBuild>false</RazorCompileOnBuild>
    <RazorCompileOnPublish>false</RazorCompileOnPublish>
    ```
- Production mode:
  - _Production mode requires you to compile your views at build/publish time [...]_

I've created a PR to update the documentation to make this more explicit: https://github.com/umbraco/UmbracoDocs/pull/4269.

Let me know if this fixes the issue for you and whether adding this note might have avoided the confusion! I also recommend watching the [Unboxing Umbraco 10.1 video](https://umbraco.com/blog/umbraco-101-release/), as I've included this gotcha in the demo 😄
